### PR TITLE
Fix InhibitionClusterIsNotRunningPrometheusAgent on CAPI

### PR DESCRIPTION
**Pull request by QuentinBisson**
Before adding a new alerting rule into this repository you should consider creating an SLO rules instead.
SLO helps you both increase the quality of your monitoring and reduce the alert noise.

* How to create a SLO rule: https://github.com/giantswarm/sloth-rules#how-to-create-a-slo
* Documentation: https://intranet.giantswarm.io/docs/monitoring/slo-alerting/

---
Towards: https://github.com/giantswarm/giantswarm/issues/29358

This PR fixes the Prometheus Agent inhibition on CAPI WCs

### Checklist

- [x] Update CHANGELOG.md
- [ ] Add [Unit tests](https://github.com/giantswarm/prometheus-rules/#testing)
- [ ] Follow [Alert structure](https://github.com/giantswarm/prometheus-rules/#how-alerts-are-structured)
- [ ] Consider [creating a dashboard](https://docs.giantswarm.io/getting-started/observability/grafana/custom-dashboards/) ([guidelines](https://intranet.giantswarm.io/docs/product/ux/guidelines/dashboards/)) (if it does not exist already) to help oncallers monitor the status of the issue.
- [x] Request review from oncall area, as well as team (e.g: `oncall-kaas-cloud` GitHub group).

### CODE REVIEW

File: helm/prometheus-rules/templates/alerting-rules/inhibit.prometheus-agent.rules.yml [hunk 1]
```diff
@@ -21,6 +21,24 @@ spec:
...
+            sum_over_time(
+              kube_namespace_created{namespace!="{{ .Values.managementCluster.name }}-prometheus", namespace=~".+-prometheus"}[5m]
+            ), "cluster_id", "$1", "namespace", "(.+)-prometheus"
+          )
+        ) by (cluster_id)
+        unless
+        count(
+          label_replace(
+            sum_over_time(
+              app_operator_app_info{app="prometheus-agent"}[5m]
```
Comment (QuentinBisson) [inline helm/prometheus-rules/templates/alerting-rules/inhibit.prometheus-agent.rules.yml:37]: Relabeling taken from https://github.com/giantswarm/prometheus-rules/blob/d3e80360943476dd21f60b461e8589995884c649/helm/prometheus-rules/templates/alerting-rules/app.rules.yml#L89

**Comment by hervenicol** [comment]
I have a simpler take: is there any reason a CAPI cluster would not have prometheus-agent running?
This inhibition was supposed to run on old clusters that have not migrated yet to a version running prometheus-agent.
Maybe we should just have a big `if not capi` around this rule?

**Comment by QuentinBisson** [comment]
Well, that's probably going to be easier to fix :D

**Comment by QuentinBisson** [comment]
Should be better now

**Review by hervenicol — APPROVED** [review]
LGTM, hope it was not hiding lots of alerts of CAPI clusters :see_no_evil:

**Comment by QuentinBisson** [comment]
Let's see :D

**Pull request merged**